internal/api/option/runtime.go

```diff
@@ -0,0 +1,37 @@
+package option
+
+import (
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+const defaultStackLength = 32
+
+func takeStacktrace(n, skip uint) string {
```

Comment:
`pkg/errors` has a way of getting stack traces already. But I think maybe you are trying to safeguard against the possibility that `pkg/errors` isn't always used? If that's the case, I understand.

https://pkg.go.dev/github.com/pkg/errors#hdr-Retrieving_the_stack_trace_of_an_error_or_wrapper

Would it be worth writing a unit test for this function?

Reply:
Yep, it's just for handling `panic` only - so most case it will not be used.

`logrus` seems not to support logging stack (https://github.com/sirupsen/logrus/issues/690) otherwise [zap](https://pkg.go.dev/go.uber.org/zap#Stack) or [zerolog](https://pkg.go.dev/github.com/rs/zerolog#Context.Stack) supports this, so I copied this source from [DataDog/dd-trace-go](https://github.com/DataDog/dd-trace-go/blob/b02945a373c04b1d0acd0a01ad55486ca3179aea/ddtrace/tracer/span.go#L313-L344). But not sure we should copy the test case too 🤔